### ISSUE TITLE
size _UNK _UNK

### ISSUE THREAD

**Issue by grostein** [llSourcell/tensorflow_chatbot#9]
I'm training the chat. More than 5 hours.
Making the test while training I have always:

```
 Reading model parameters from working_dir/seq2seq.ckpt-12850
 > hello
size _UNK _UNK
```

I'm doing something wrong?
I'm at:
`global step 12850 learning rate 0.3589 step-time 0.44 perplexity 1.00`

**Comment by jayasuryajsk** [comment]
Even i have the same problem

**Comment by grostein** [comment]
There is no solution I suppose. I can't find anything in the code.

**Comment by Seiwert** [comment]
So I was having this problem as well. I did some digging and I found this comment on the YouTube video:
![image](https://cloud.githubusercontent.com/assets/4229602/21000275/8224e364-bcdf-11e6-9a55-6ef85aaa1353.png)

I just modified my seq2seq.ini file to reflect this. I'll let you know if this helps the issue.

**Comment by HazeHub** [comment]
Am trying too, will let you know

**Comment by Seiwert** [comment]
Heh, wow. So I'm an idiot...

I don't know about your guys setup, but you may want to double check your test/train .enc/.dec files in your data directory. If you screwed up like I did, then you copied directly from this guys repo: https://github.com/suriyadeepan/datasets/tree/master/seq2seq/cornell_movie_corpus/processed_data and just took the already processed data. This will not work...

If your train and test files look like this, then you're going to have a bad time: 
![image](https://cloud.githubusercontent.com/assets/4229602/21029063/98673f7c-bd5d-11e6-852e-622f4cf19412.png)

They should look like this:
![image](https://cloud.githubusercontent.com/assets/4229602/21029217/41f158ca-bd5e-11e6-989b-60e3fa4b0087.png)

To get them to look like that, you need to do what this guy said:
![image](https://cloud.githubusercontent.com/assets/4229602/21029253/6c0aceac-bd5e-11e6-8c28-90260c3a06e8.png)

However, he missed a step. You need to copy the prepare_data.py file into the data directory you created which contains the cornell movie scripts. 

Then, when you run it again, you should be seeing this in your terminal:
![image](https://cloud.githubusercontent.com/assets/4229602/21029307/b31e27b2-bd5e-11e6-9cdc-684345c1f8a8.png)

I believe this is everyone's issue. Reason being is because I got to step 95k and had a perplexity score of 1.00. As you can see, my perplexity score is around 60 and started at 366. This makes a lot more sense now that the dataset it's learning from is much larger. I'm going to let this run for a while and will let you know if this fixes it.

**Comment by Seiwert** [comment]
So I've been running for about 8 hours now since fixing my train and test files and it appears to be working for me. I still occasionally get "_UNK" but it's not nearly as bad as it was before. For other peoples reference, I'm at global step 45,600 with perplexity of 10.62 and this is giving okay responses.
![image](https://cloud.githubusercontent.com/assets/4229602/21038303/18736af6-bd9a-11e6-9976-58aff9f696f9.png)

Once again, these results took about 8 hours for me on an AMD-9590FX.

**Comment by HazeHub** [comment]
@Seiwert you made my day. Such a small dataset would explain all the unknown tokens

**Comment by umasslowellmj** [comment]
I thought I followed everything here, and trained the model, still getting white space or _UNK's. 
Can anyone verify if I'm doing everything correctly.
 - Uncommented the last line of prepare_data.py and ran it,
 - moved the outputs (lot of train.dec, test.dec, train.dec.ids20000, etc files) under tensorflow_chatbot/data ,
 - checked those .dec files contain the texts, 
 - ran the training till perplexity became 40 something, 

and tested: 
>> Mode : test

Reading model parameters from working_dir/seq2seq.ckpt-275700
hi
> _UNK .
> hello
_UNK .
>

**Comment by Seiwert** [comment]
@umasslowellmj That all seems to be correct. It is very odd that your perplexity is still 40 after running it to that many steps. 

If you would like, I actually have some trained models fully uploaded at my repository here: https://github.com/Seiwert/chatBot
This repo also integrates with Alexa if you're interested in that.

Simply download whatever iteration you want from the working_dir folder and modify your checkpoint file to look at that iteration. 

One thing that may have been your issue is if you didn't modify your network and layer size. In my comment above, I have a screenshot of a youtube comment saying to change network size to 128 and layer size to 1. This will be in your seq2seq.ini file. 

Hope this helps!

**Comment by umasslowellmj** [comment]
@Seiwert  oh yea, I had been running the training for couple days and felt like something was wrong, and then I found this thread. thats right i need to delete the working_dir completely and start over or could just use your trained model. Thanks, it helps a lot.

**Comment by steve3nto** [comment]
I get this error when running the prepare_data.py

'utf-8' codec can't decode byte 0xad in position 585399: invalid start byte

what shall I do to create the train and test files properly?

**Comment by ugdesmond** [comment]
@steve3nto go to your vocab2000.enc and vocab200.dec delete those words with("?") or  write a code that will ignore if not utf-8

**Comment by lohith-emplay** [comment]
There are no "?" in the vocab2000.enc and vocab200.dec  files. can't seem to fix the problem.
